### PR TITLE
Create adaptations subtype of Sprite Lab

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Creates a standalone "Adaptations" sub-type of Sprite Lab. Primarily followed the example in [this PR](https://github.com/code-dot-org/code-dot-org/pull/49975). For the new level file, I just cloned the given example level on my localhost environment, and named it "New Adaptations Project" - let me know if there's anything else I need to do here! 

## Links

https://codedotorg.atlassian.net/browse/SL-728

## Testing story

Tested by navigating to `projects/adapations/new` and confirming that this creates a new project using the Adaptations level definition and block pools. Also confirmed that the block pool and toolbox matches the [example level](https://levelbuilder-studio.code.org/s/csc-adaptations-2023/lessons/3/levels/8). Let me know if there's any other ways I should test this!

### CODE REVIEW

File: dashboard/config/scripts/levels/New Adaptations Project.level [hunk 1]
```diff
@@ -0,0 +1,1553 @@
...
+    "auto_run_setup": "DRAW_LOOP",
+    "parent_level_id": 15503,
+    "callout_json": "[\r\n {\r\n  \"localization_key\": \"behaviors_edit\",\r\n  \"callout_text\": \"Add code to this behavior.\",\r\n  \"element_id\": \"#callout\",\r\n  \"on\": \"\",\r\n  \"qtip_config\": {\r\n   \"codeStudio\": {\r\n    \"canReappear\": false,\r\n    \"dropletPaletteCategory\": \"\"\r\n   },\r\n   \"style\": {\r\n    \"classes\": \"\"\r\n   },\r\n   \"position\": {\r\n    \"my\": \"top right\",\r\n    \"at\": \"bottom right\",\r\n    \"adjust\": {\r\n     \"x\": -15,\r\n     \"y\": 0\r\n    }\r\n   }\r\n  }\r\n }\r\n]",
+    "encrypted": "false",
+    "mini_rubric": "false",
+    "encrypted_examples": "fMF3FV5MzXOakqxo8NXETKQpLfkvPHEx9araR1dB/VZWPnXbcpo2k2VUMUJb\nRGL9yvx3ji20YjjOnS1PCLEdCw==\n",
+    "validation_enabled": "false",
+    "mini_toolbox": "false",
+    "hide_pause_button": "false",
+    "standalone_app_name": "spritelab",
```
Comment (molly-moen) [inline dashboard/config/scripts/levels/New Adaptations Project.level]: the standalone_app_name here should be 'adaptations'

**Review by molly-moen — COMMENTED** [review]
Looks good! Another couple things to test:
- Can you create a level in levelbuilder with the adaptations subtype?
- When you share/remix that level, does it get shared/remixed to the right url and against the 'New Adaptations Project' level?
- Does the project name, share and remix show up in the header for the standalone project? (sometimes I had issues with this not being there and needed to modify the level config)

File: dashboard/config/scripts/levels/New Adaptations Project.level (hunk 1 above)
Comment (sanchitmalhotra126) [reply dashboard/config/scripts/levels/New Adaptations Project.level]: Oh good catch! updated

**Comment by sanchitmalhotra126** [comment]
> * Can you create a level in levelbuilder with the adaptations subtype?

Yep, looks like it -

Screenshot 2023-04-27 at 3 49 01 PM

> * When you share/remix that level, does it get shared/remixed to the right url and against the 'New Adaptations Project' level?

Yep, when remixing a new adaptations level, I get (for example) http://localhost-studio.code.org:3000/projects/adaptations/kxBT1mmOrh-Ff1HIG0Xa2A/edit

> * Does the project name, share and remix show up in the header for the standalone project? (sometimes I had issues with this not being there and needed to modify the level config)

Ah good call, the header is blank. I'll double check the level config in your PR and update this.

Edit: updated to show Share, Remix, and project name:
Screenshot 2023-04-27 at 6 09 16 PM

**Comment by fisher-alice** [comment]
> Looks good! Another couple things to test:
> 
> * Can you create a level in levelbuilder with the adaptations subtype?
> * When you share/remix that level, does it get shared/remixed to the right url and against the 'New Adaptations Project' level?
> * Does the project name, share and remix show up in the header for the standalone project? (sometimes I had issues with this not being there and needed to modify the level config)

Could you also check if you edit an animation, are you able to remix the project? Thanks!

File: dashboard/config/scripts/levels/New Adaptations Project.level [hunk 2]
```diff
@@ -23,7 +23,7 @@
     "disable_procedure_autopopulate": "false",
     "top_level_procedure_autopopulate": "false",
     "use_modal_function_editor": "true",
-    "hide_share_and_remix": "true",
+    "hide_share_and_remix": "false",
```
Comment (molly-moen) [inline dashboard/config/scripts/levels/New Adaptations Project.level:26]: we should probably document these flags needed for a standalone project level somewhere 🤔

Comment (sanchitmalhotra126) [reply dashboard/config/scripts/levels/New Adaptations Project.level:26]: Yeah, i think I just had to set "hide_share_and_remix" to false and "is_project_level" to true; the first was configurable through levelbuilder but the second had to be added manually, not sure if there's a way to configure that on level creation?

**Comment by sanchitmalhotra126** [comment]
> Could you also check if you edit an animation, are you able to remix the project? Thanks!

I believe the animation tab is disabled? At least, that's how the example level that was cloned was set up.

File: dashboard/config/scripts/levels/New Adaptations Project.level (hunk 2 above)
Comment (mikeharv) [reply dashboard/config/scripts/levels/New Adaptations Project.level:26]: +1 - I would assume we'd also want `"hide_animation_mode": "false",` for any project level. This is configurable through levelbuilder and adds the Costumes/Backgrounds tabs (Backgrounds currently require an experiment flag).

**Comment by mikeharv** [comment]
> > Could you also check if you edit an animation, are you able to remix the project? Thanks!
> 
> I believe the animation tab is disabled? At least, that's how the example level that was cloned was set up.

@fisher-alice  I'm guessing you meant to say "upload an animation," not just edit one? @sanchitmalhotra126 if you uncheck the box to the hide the animations tab, you should be able to test this behavior locally, even if you end up not committing the change to the level file for whatever reason.

**Review by mikeharv — APPROVED** [review]
I'll defer to @molly-moen for final approval, but this looks good to me! I don't think the decision to hide or show the animation tabs is a blocker here because the curriculum team can toggle this themselves. It also seems unlikely that the adding a new subtype would cause any kind of issue with the student upload work.

**Review by molly-moen — APPROVED** [review]
LGTM! I also think we can turn on the animations tab, but it's not a blocker

File: dashboard/config/scripts/levels/New Adaptations Project.level (hunk 2 above)
Comment (sanchitmalhotra126) [reply dashboard/config/scripts/levels/New Adaptations Project.level:26]: Ah gotcha, sounds good. Just enabled that via levelbuilder.

**Pull request merged**